PostHog/PostHogSDK.swift

```diff
@@ -844,15 +852,17 @@ let maxRetryDelay = 30.0
     }
 
     private func reportFeatureFlagCalled(flagKey: String, flagValue: Any?) {
-        if !flagCallReported.contains(flagKey) {
-            let properties: [String: Any] = [
-                "$feature_flag": flagKey,
-                "$feature_flag_response": flagValue ?? "",
-            ]
+        featureFlagsLock.withLock {
+            if !flagCallReported.contains(flagKey) {
+                let properties: [String: Any] = [
+                    "$feature_flag": flagKey,
+                    "$feature_flag_response": flagValue ?? "",
+                ]
 
-            flagCallReported.insert(flagKey)
+                flagCallReported.insert(flagKey)
 
-            capture("$feature_flag_called", properties: properties)
+                capture("$feature_flag_called", properties: properties)
```

Comment:
I'd move this outside of the lock, the only concurrent part is the mutable `flagCallReported`